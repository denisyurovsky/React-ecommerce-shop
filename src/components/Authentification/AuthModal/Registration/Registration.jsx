import { Box, Button } from '@mui/material/';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

import {
  MIN_PASSWORD_LENGTH,
  authStatus,
} from '../../../../constants/authConstants.js';
import checkEmailValidity from '../../../../helpers/checkEmailValidity.js';
import checkPasswordValidity from '../../../../helpers/checkPasswordValidity.js';
import LoadingButton from '../../../ui-kit/buttons/LoadingButton';
import { StandartedTextField } from '../../../ui-kit/StandartedTextField/StandartedTextField.jsx';
import PasswordForm from '../PasswordForm';
import useErrorHandler from '../useErrorHandler.js';

const { PENDING, FULFILLED, REJECTED, IDLE } = authStatus;

const Registration = ({ sendForm, errorMessage, status }) => {
  const isLoading = status === PENDING;
  const isError = status === REJECTED;
  const { isError: isServerError, resetError } = useErrorHandler(isError);

  const [values, setValues] = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
  });

  const [confirmPassword, setConfirmPassword] = useState('');
  const [showPassword, setShowPassword] = useState(false);

  const [checkValues, setCheckValues] = useState({
    isEmailValid: true,
    isPasswordValid: true,
    isConfirmPasswordValid: true,
  });

  const [isFinished, setIsFinished] = useState(false);

  const checkName = (name) => name.length > 0;
  const checkEmail = (email) => email.match(/@/);
  const checkPassword = (password) => password.length > MIN_PASSWORD_LENGTH;
  const checkMatch = (str1, str2) => str1 === str2;
  const checkIsFinished = ({
    firstName,
    lastName,
    email,
    password,
    confirmPassword,
  }) =>
    checkName(firstName) &&
    checkName(lastName) &&
    checkEmail(email) &&
    checkPassword(password) &&
    checkPassword(confirmPassword);

  const checkValidForm = ({ email, password, confirmPassword }) => {
    const isEmailValid = checkEmailValidity(email);
    const isPasswordValid = checkPasswordValidity(password);
    const isConfirmPasswordValid = checkMatch(password, confirmPassword);

    setCheckValues({
      isEmailValid,
      isPasswordValid,
      isConfirmPasswordValid,
    });

    return isEmailValid && isPasswordValid && isConfirmPasswordValid;
  };

  const preparetoSend = () => {
    if (checkValidForm({ ...values, confirmPassword })) {
      sendForm({ ...values });
    }
  };

  const onFirstNameChange = (e) => {
    const firstName = e.target.value;

    setIsFinished(checkIsFinished({ ...values, firstName, confirmPassword }));
    setValues({ ...values, firstName });
  };

  const onLastNameChange = (e) => {
    const lastName = e.target.value;

    setIsFinished(checkIsFinished({ ...values, lastName, confirmPassword }));
    setValues({ ...values, lastName });
  };
  const onEmailChange = (e) => {
    const email = e.target.value;

    setIsFinished(checkIsFinished({ ...values, email, confirmPassword }));
    resetError();
    setValues({ ...values, email });
  };
  const onPasswordChange = (e) => {
    const password = e.target.value;

    setIsFinished(checkIsFinished({ ...values, password, confirmPassword }));
    setValues({ ...values, password: e.target.value });
  };
  const onConfirmPasswordChange = (e) => {
    const confirmPassword = e.target.value;

    setIsFinished(checkIsFinished({ ...values, confirmPassword }));
    setConfirmPassword(confirmPassword);
  };
  const onVisibilityChange = () => setShowPassword(!showPassword);

  const passwordError =
    !checkValues.isConfirmPasswordValid || !checkValues.isPasswordValid;

  const isEmailWrong = isServerError || !checkValues.isEmailValid;
  const emailMessage = isServerError ? errorMessage : 'Email is invalid';

  return (
    <Box>
      <StandartedTextField
        value={values.firstName}
        labelText="First name"
        onChange={onFirstNameChange}
      />
      <StandartedTextField
        labelText="Last name"
        value={values.lastName}
        onChange={onLastNameChange}
      />
      <StandartedTextField
        labelText="Email"
        value={values.email}
        onChange={onEmailChange}
        error={isEmailWrong}
        helperText={isEmailWrong ? emailMessage : null}
      />
      <PasswordForm
        password={values.password}
        onChange={onPasswordChange}
        isError={passwordError}
        errorMessage={
          !checkValues.isConfirmPasswordValid
            ? "Passwords don't match"
            : 'Should contain upper case letters, special symbols, digits and be at least 8 symbols in length.'
        }
        hidden={!showPassword}
        onHiddenChange={onVisibilityChange}
      />
      <PasswordForm
        label="Confirm password"
        password={confirmPassword}
        onChange={onConfirmPasswordChange}
        isError={!checkValues.isConfirmPasswordValid}
        errorMessage="Passwords don't match"
        hidden={!showPassword}
        onHiddenChange={onVisibilityChange}
      />
      <Box sx={{ mt: 4 }} />
      {isLoading ? (
        <LoadingButton label="Loading..." isFullWidth={true} />
      ) : (
        <Button
          fullWidth
          variant="contained"
          onClick={preparetoSend}
          disabled={!isFinished}
        >
          Sign up
        </Button>
      )}
    </Box>
  );
};

Registration.defaultProps = {
  status: IDLE,
  errorMessage: '',
};

Registration.propTypes = {
  sendForm: PropTypes.func.isRequired,
  status: PropTypes.oneOf([PENDING, FULFILLED, REJECTED, IDLE]),
  errorMessage: PropTypes.string,
};

export default Registration;
