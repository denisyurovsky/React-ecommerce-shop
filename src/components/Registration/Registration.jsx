import { Visibility, VisibilityOff } from '@mui/icons-material';
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import FormControl from '@mui/material/FormControl';
import IconButton from '@mui/material/IconButton';
import Input from '@mui/material/Input';
import InputAdornment from '@mui/material/InputAdornment';
import InputLabel from '@mui/material/InputLabel';
import TextField from '@mui/material/TextField';
import React, { useState } from 'react';

import isEmailValid from '../../helpers/isEmailValid';
import isPasswordValid from '../../helpers/isPasswordValid';

import styles from './Registration.module.scss';

const Registration = () => {
  const [values, setValues] = useState({
    showPassword: false,
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: '',
    isEmailValid: true,
    isPasswordValid: true,
    isConfirmPasswordValid: true,
    isFirstNameValid: true,
    isLastNameValid: true,
  });

  const validateForm = (
    email,
    password,
    confirmPassword,
    firstName,
    lastName
  ) => {
    const isFirstNameOk = firstName.length > 0;
    const isLastNameOk = lastName.length > 0;
    const isEmailOk = isEmailValid(email);
    const isPasswordOk = isPasswordValid(password);
    const isConfirmPasswordOk = password === confirmPassword;

    return setValues({
      ...values,
      isEmailValid: isEmailOk,
      isPasswordValid: isPasswordOk,
      isConfirmPasswordValid: isConfirmPasswordOk,
      isFirstNameValid: isFirstNameOk,
      isLastNameValid: isLastNameOk,
    });
  };

  const handleClickShowPassword = () => {
    setValues({
      ...values,
      showPassword: !values.showPassword,
    });
  };

  const onFirstNameChange = (e) => {
    setValues({ ...values, firstName: e.target.value });
  };
  const onLastNameChange = (e) => {
    setValues({ ...values, lastName: e.target.value });
  };
  const onEmailChange = (e) => {
    setValues({ ...values, email: e.target.value });
  };
  const onPasswordChange = (e) => {
    setValues({ ...values, password: e.target.value });
  };
  const onConfirmPasswordChange = (e) => {
    setValues({ ...values, confirmPassword: e.target.value });
  };

  return (
    <Container
      className={styles.containerWrapper}
      maxWidth="sm"
      sx={{ m: 'auto' }}
    >
      <div className={styles.registrationContainer}>
        <TextField
          data-testid="firstNameInput"
          error={!values.isFirstNameValid}
          onChange={onFirstNameChange}
          label="First name"
          variant="standard"
        />
        <TextField
          data-testid="lastNameInput"
          error={!values.isLastNameValid}
          onChange={onLastNameChange}
          label="Last name"
          variant="standard"
        />
        <TextField
          data-testid="emailInput"
          onChange={onEmailChange}
          error={!values.isEmailValid}
          label="Email"
          variant="standard"
        />
        <FormControl variant="standard">
          <InputLabel
            error={!values.isPasswordValid}
            htmlFor="standard-adornment-password"
          >
            Password
          </InputLabel>
          <Input
            data-testid="passwordInput"
            type={values.showPassword ? 'text' : 'password'}
            onChange={onPasswordChange}
            error={!values.isPasswordValid}
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  data-testid="hiddenButton"
                  aria-label="toggle password visibility"
                  onClick={handleClickShowPassword}
                >
                  {values.showPassword ? <VisibilityOff /> : <Visibility />}
                </IconButton>
              </InputAdornment>
            }
          />
        </FormControl>
        <FormControl variant="standard">
          <InputLabel
            error={!values.isConfirmPasswordValid}
            htmlFor="standard-adornment-password"
          >
            Confirm Password
          </InputLabel>
          <Input
            data-testid="confirmPasswordInput"
            onChange={onConfirmPasswordChange}
            error={!values.isConfirmPasswordValid}
            type={values.showPassword ? 'text' : 'password'}
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  aria-label="toggle password visibility"
                  onClick={handleClickShowPassword}
                >
                  {values.showPassword ? <VisibilityOff /> : <Visibility />}
                </IconButton>
              </InputAdornment>
            }
          />
        </FormControl>
        <Button
          data-testid="signUpButton"
          type="submit"
          fullWidth
          variant="contained"
          sx={{ mt: 3, mb: 2 }}
          onClick={() =>
            validateForm(
              values.email,
              values.password,
              values.confirmPassword,
              values.firstName,
              values.lastName
            )
          }
        >
          Sign Up
        </Button>
      </div>
    </Container>
  );
};

export default Registration;
