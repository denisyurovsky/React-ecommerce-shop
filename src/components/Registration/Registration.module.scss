.containerWrapper {
  position: relative;
}

.registrationContainer {
  display: flex;
  width: 90%;
  margin: auto;
  gap: 25px;
  flex-direction: column;
}

.closeIcon {
  position: absolute;
  top: 10px;
  right: 10px;
}
