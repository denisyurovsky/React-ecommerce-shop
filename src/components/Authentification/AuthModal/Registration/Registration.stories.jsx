import React from 'react';

import Registration from './Registration';

export default {
  title: 'AuthModal/Registration',
  component: Registration,
  argTypes: { sendForm: { action: 'Sending form' } },
};

const Template = (args) => <Registration {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Error = Template.bind({});
Error.storyName = 'Server error';
Error.args = {
  status: 'failed',
  errorMessage: 'User already exists',
};

export const Loading = Template.bind({});
Loading.args = {
  status: 'loading',
};
