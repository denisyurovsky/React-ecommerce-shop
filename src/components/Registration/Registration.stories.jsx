import React from 'react';

import Registration from './Registration';

export default {
  title: 'Registration',
  component: Registration,
  argTypes: {},
};

const Template = (args) => <Registration {...args} />;

export const RegistrationMain = Template.bind({});
RegistrationMain.args = {};
