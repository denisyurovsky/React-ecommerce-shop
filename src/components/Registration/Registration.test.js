import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import Registration from './Registration';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Registration />);

  expect(asFragment()).toMatchSnapshot();
});

describe('toggle password input', () => {
  it('toggles password input', () => {
    render(<Registration />);

    const toggleButton = screen.getByTestId('hiddenButton');

    const passwordInput = screen
      .getByTestId('passwordInput')
      .querySelector('input');

    userEvent.click(toggleButton);

    expect(passwordInput.getAttribute('type')).toBe('text');

    userEvent.click(toggleButton);

    expect(passwordInput.getAttribute('type')).toBe('password');
  });
});

describe('Successful registration scenario', () => {
  it('should log success message', () => {
    render(<Registration />);

    const firstNameInput = screen
      .getByTestId('firstNameInput')
      .querySelector('input');

    fireEvent.change(firstNameInput, { target: { value: 'Denis' } });

    const lastNameInput = screen
      .getByTestId('lastNameInput')
      .querySelector('input');

    fireEvent.change(lastNameInput, { target: { value: 'Iurovskii' } });

    const emailInput = screen.getByTestId('emailInput').querySelector('input');

    fireEvent.change(emailInput, { target: { value: 'denis@epam.com' } });

    const passwordInput = screen
      .getByTestId('passwordInput')
      .querySelector('input');

    fireEvent.change(passwordInput, { target: { value: 'AbcDefG12#' } });

    const confirmPasswordInput = screen
      .getByTestId('confirmPasswordInput')
      .querySelector('input');

    fireEvent.change(confirmPasswordInput, { target: { value: 'AbcDefG12#' } });

    userEvent.click(screen.getByTestId('signUpButton'));

    expect(firstNameInput.getAttribute('aria-invalid')).toBe('false');

    expect(lastNameInput.getAttribute('aria-invalid')).toBe('false');

    expect(emailInput.getAttribute('aria-invalid')).toBe('false');

    expect(passwordInput.getAttribute('aria-invalid')).toBe('false');

    expect(confirmPasswordInput.getAttribute('aria-invalid')).toBe('false');
  });
});

describe('failed registration scenarios', () => {
  describe('when first name is empty', () => {
    it('should not be empty', () => {
      render(<Registration />);

      const firstNameInput = screen
        .getByTestId('firstNameInput')
        .querySelector('input');

      fireEvent.change(firstNameInput, { target: { value: '' } });

      userEvent.click(screen.getByTestId('signUpButton'));

      expect(firstNameInput.getAttribute('aria-invalid')).toBe('true');
    });
  });
  describe('when last name is empty', () => {
    it('should not be empty', () => {
      render(<Registration />);

      const lastNameInput = screen
        .getByTestId('lastNameInput')
        .querySelector('input');

      fireEvent.change(lastNameInput, { target: { value: '' } });

      userEvent.click(screen.getByTestId('signUpButton'));

      expect(lastNameInput.getAttribute('aria-invalid')).toBe('true');
    });
  });

  describe('when email is invalid', () => {
    it('should be valid', () => {
      render(<Registration />);

      const emailInput = screen
        .getByTestId('emailInput')
        .querySelector('input');

      fireEvent.change(emailInput, { target: { value: 'denis' } });

      userEvent.click(screen.getByTestId('signUpButton'));

      expect(emailInput.getAttribute('aria-invalid')).toBe('true');
    });
  });

  describe('when password is invalid', () => {
    it('should be valid', () => {
      render(<Registration />);

      const passwordInput = screen
        .getByTestId('passwordInput')
        .querySelector('input');

      fireEvent.change(passwordInput, { target: { value: '12345678' } });

      userEvent.click(screen.getByTestId('signUpButton'));

      expect(passwordInput.getAttribute('aria-invalid')).toBe('true');
    });
  });
  describe('when confirm password does not match password', () => {
    it('should be same as password', () => {
      render(<Registration />);

      const passwordInput = screen
        .getByTestId('passwordInput')
        .querySelector('input');

      const confirmPasswordInput = screen
        .getByTestId('confirmPasswordInput')
        .querySelector('input');

      fireEvent.change(passwordInput, { target: { value: 'MoonWave$4' } });

      fireEvent.change(confirmPasswordInput, {
        target: { value: 'MoonWave$' },
      });

      userEvent.click(screen.getByTestId('signUpButton'));

      expect(confirmPasswordInput.getAttribute('aria-invalid')).toBe('true');
    });
  });
});
