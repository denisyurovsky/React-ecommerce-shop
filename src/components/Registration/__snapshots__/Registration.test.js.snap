// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm containerWrapper css-dj5xlo-MuiContainer-root"
  >
    <div
      class="registrationContainer"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        data-testid="firstNameInput"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-1"
          id="mui-1-label"
        >
          First name
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="mui-1"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        data-testid="lastNameInput"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-2"
          id="mui-2-label"
        >
          Last name
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="mui-2"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        data-testid="emailInput"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-3"
          id="mui-3-label"
        >
          Email
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="mui-3"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="standard-adornment-password"
        >
          Password
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="passwordInput"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            type="password"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="hiddenButton"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="standard-adornment-password"
        >
          Confirm Password
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1480iag-MuiInputBase-root-MuiInput-root"
          data-testid="confirmPasswordInput"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            type="password"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
        data-testid="signUpButton"
        tabindex="0"
        type="submit"
      >
        Sign Up
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
